packages/fiori/src/IllustratedMessage.js

```diff
@@ -0,0 +1,245 @@
+ * a <code>ui5-dialog</code> or a <code>ui5-page</code>
+ *
+ * For the <code>ui5-illustrated-message</code>
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents-fiori/dist/IllustratedMessage.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.fiori.IllustratedMessage
+ * @extends UI5Element
+ * @tagname ui5-illustrated-message
+ * @public
+ * @since 1.0.0-rc.15
+ */
+class IllustratedMessage extends UI5Element {
+	constructor() {
+		super();
+
+		this.i18nBundle = getI18nBundle("@ui5/webcomponents-fiori");
+		this._handleResize = this.handleResize.bind(this);
+	}
+
+	static get metadata() {
+		return metadata;
+	}
+
+	static get BREAKPOINTS() {
+		return {
+			DIALOG: 679,
+			SPOT: 319,
+			BASE: 259,
+		};
+	}
+
+	static get MEDIA() {
+		return {
+			BASE: "base",
+			SPOT: "spot",
+			DIALOG: "dialog",
+			SCENE: "scene",
+		};
+	}
+
+	onBeforeRendering() {
+		const illustrationData = getIllustrationDataSync(this.name);
+
+		if (illustrationData === ILLUSTRATION_NOT_FOUND) {
+			this.invalid = true;
+			/* eslint-disable-next-line */
+			return console.warn(`Required illustration is not registered. You can either import the illustration as a module in order to use it e.g. "@ui5/webcomponents-fiori/dist/illustrations/${this.name}.js".`);
+		}
+
+		this.invalid = false;
+		this.spotSvg = illustrationData.spotSvg;
+		this.dialogSvg = illustrationData.dialogSvg;
+		this.sceneSvg = illustrationData.sceneSvg;
+
+		this.illustrationTitle = this.i18nBundle.getText(illustrationData.title);
+		this.illustrationSubtitle = this.i18nBundle.getText(illustrationData.subtitle);
+	}
+
+	onEnterDOM() {
+		ResizeHandler.register(this, this._handleResize);
+	}
+
+	onExitDOM() {
+		ResizeHandler.deregister(this, this._handleResize);
+	}
+
+	handleResize() {
+		if (this.offsetWidth <= IllustratedMessage.BREAKPOINTS.BASE) {
+			this.media = IllustratedMessage.MEDIA.BASE;
+		} else if (this.offsetWidth <= IllustratedMessage.BREAKPOINTS.SPOT) {
+			this.media = IllustratedMessage.MEDIA.SPOT;
+		} else if (this.offsetWidth <= IllustratedMessage.BREAKPOINTS.DIALOG) {
+			this.media = IllustratedMessage.MEDIA.DIALOG;
+		} else {
+			this.media = IllustratedMessage.MEDIA.SCENE;
+		}
+	}
+
+	get effectiveIllustration() {
+		switch (this.media) {
+		case IllustratedMessage.MEDIA.SPOT:
+			return this.spotSvg;
+		case IllustratedMessage.MEDIA.DIALOG:
+			return this.dialogSvg;
+		case IllustratedMessage.MEDIA.SCENE:
+			return this.sceneSvg;
+		default:
+			return "";
+		}
+	}
+
+	get effectiveTitleText() {
+		return this.titleText ? this.titleText : this.illustrationTitle;
+	}
+
+	get effectiveSubitleText() {
+		return this.subtitleText ? this.subtitleText : this.illustrationSubtitle;
+	}
+
+	get hasActions() {
+		return !!this.actions.length && this.media !== IllustratedMessage.MEDIA.BASE;
+	}
+
+	static get render() {
+		return litRender;
+	}
```

Comment:
Really minor, but still it's good to have static get render, styles, template and metadata at one place grouped. Maybe you can move these ones between `static get metadata` and `static get BREAKPOINTS`